hook.js

```diff
@@ -9,68 +10,75 @@ var parseSource = require('./sourceParse');
 
 var token = 'token ' + process.env.OAUTH_TOKEN;
 var productName = 'Discord';
+var githubClient = github.client();
 
-var comment = function(request, response) {
-    var payload = request.body;
-    var configMetadataURL = payload.repository.contents_url.replace('{+path}', '.doiuse');
+function handle(request, response) {
+    var eventType = request.headers['x-github-event'];
+    var metadata = request.body;
 
+    // React to pull requests only
+    if(eventType === 'pull_request') {
+       response.status(200).send('OK');
 
-    response.status(200).send('ok');
-
-    // We only support pull requests at this time
-    if (request.headers['x-github-event'] !== 'pull_request') {
-        return;
+       printDebug(metadata);
+       comment(metadata);
     }
+}
 
-    // TODO: Only acknowledge pushes to the "Master" branch.
-    console.log('Repo: ' + payload.repository.full_name);
-    console.log('Payload:');
-    console.log(payload);
+function comment(metadata) {
+    var pullRequestNumber = metadata.number;
+    var destinationRepository = metadata.repository.full_name;
+    var originRepository = metadata.pull_request.head.repo.full_name;
+    var originBranch = metadata.pull_request.head.ref;
 
-    sendRequest({
-        url: configMetadataURL,
-        headers: {
-            'User-Agent': productName
-        }
-    }, function(error, response, body) {
-        var configMetadata = JSON.parse(body);
-        var config = ['last 2 versions'];
-        var commitsURL = payload.pull_request.commits_url;
-        var commentURL = payload.pull_request.review_comments_url;
-        var configMetadataContent;
-
-        if (configMetadata.content) {
-            configMetadataContent = new Buffer(configMetadata.content, 'base64').toString();
-
-            // Consider text separated by commas and linebreaks to be individual
-            // options
-            config = configMetadataContent.replace(/\r?\n|\r/g, ', ').split(/,\s*/);
-        }
+    trackUsage(originRepository);
+
+    getConfiguration(originRepository, originBranch, function(config) {
+        var prClient = githubClient.pr(destinationRepository, pullRequestNumber);
+
+        prClient.commits(function(error, commits) {
+            if(error) return console.error(error);
```

Comment:
We should create a function, like `log`, for all errors, where we give them a descriptive title, like we did with regular log statements.